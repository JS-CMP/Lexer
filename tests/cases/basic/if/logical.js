if (true && false) {
    console.log("Should not run");
} else if (true || false) {
    console.log("Should run");
}