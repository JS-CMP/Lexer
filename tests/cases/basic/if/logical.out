#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {
    if (JS::Any(true) && JS::Any(false)) 
    {
        console[u"log"](JS::Any(u"Should not run"));
    } else if (JS::Any(true) || JS::Any(false)) 
    {
        console[u"log"](JS::Any(u"Should run"));
    }
    return 0;
}
